### PR TITLE
Season stats agb

### PR DESCRIPTION
**Pull request by andrew-bingham1**
Changes the game_teams.rb attribute of season_id to allow methods to check just that
Adds game_stats class w/ init method and test 
Adds best and worst coaches methods and tests

### CODE REVIEW

File: lib/game_teams.rb [hunk 1]
```diff
@@ -16,7 +16,7 @@ def initialize(details)
     @tackles = details[:tackles]
     @team_id = details[:team_id]
     @game_id = details[:game_id]
-    @season_id = @game_id[0..3]
+    @season_id = "#{@game_id[0..3].to_i}#{@game_id[0..3].to_i.next}"
```
Comment (thawley2) [inline lib/game_teams.rb:19]: Fantastic idea, that cut down my test for ONE test by like 10 min!!!!!

File: lib/season_stats.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+class SeasonStats
+  attr_reader :teams, :game_teams
+  
+  def initialize
+    @teams = CSV.open('./data/teams.csv', headers: true, header_converters: :symbol).map { |row| Team.new(row) }
+    @game_teams = CSV.open('./data/game_teams.csv', headers: true, header_converters: :symbol).map { |row| GameTeams.new(row) }
+  end
+
+  ###not pretty but it works, good opportunity to refactor###
+  def winningest_coach(season_year)
```
Comment (thawley2) [inline lib/season_stats.rb:13]: Definitely need to refactor but great job creating this

File: lib/season_stats.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    total_game_hash = Hash.new(0)
```
Comment (thawley2) [inline lib/season_stats.rb:14]: Could be a helper method

File: lib/season_stats.rb [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+          win_coach_hash[game.coach] += 1
+        end
+      end
+    end
+    total_game_hash.merge!(win_coach_hash) {|coach, games, wins| wins.to_f / games}
+    total_game_hash.key(total_game_hash.values.max)
+  end
+  
+###same as the one above, ripe for refactor###
+  def worst_coach(season_year)
```
Comment (thawley2) [inline lib/season_stats.rb:34]: See above comment

**Review by thawley2 — APPROVED** [review]
Great job, we will have to revisit slimming these bad boys down.

**Review by CrowRising — APPROVED** [review]
These changes look great! I look forward to seeing this all come together.

**Review by angelbyun — APPROVED** [review]
Great job on getting through the methods for season stats class!

**Comment by CrowRising** [comment]
Super genius fix on that @season_id to the string interpolation

**Pull request merged**